src/main/java/com/codeup/adlister/controllers/DeleteAdServlet.java

```diff
@@ -18,18 +18,18 @@
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
 
 
-
-        long id = Long.parseLong(request.getParameter("id"));
-        Ad ad = DaoFactory.getAdsDao().getAd(id);
-        DaoFactory.getAdsDao().deleteAd(ad);
-
         if (request.getSession().getAttribute("user") != null) {
             response.sendRedirect("/profile");
             return;
         } else {
             response.sendRedirect("/index");
```

Comment:
This redirect should be in line `31`. There's no need for an `else` in this `if`.